app/Services/Api/ApiEntityService.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Services\Api;
+
+use App\Models\MiscModel;
+use App\Models\Campaign;
+
+class ApiEntityService
+{
+    /**
+     */
+    public function saveEntity(array $entity, MiscModel $class, Campaign $campaign)
+    {
+        $tags = [];
+
+        // Prepare the data
+        unset($entity['module']);
+        if (!empty($entity['entry'])) {
+            $entity['entry'] = nl2br($entity['entry']);
+        }
+        if (isset($entity['tags'])) {
+            $tags = $entity['tags'];
+        }
+        request()->merge(['tags' => $tags]);
```

Comment:
Don't access the request() object in the service, as that can't be tested in the future.